development/tools/cmd/usersmapchecker/main.go

```diff
@@ -58,98 +56,60 @@ import (
 	if err != nil {
-		contextLogger.LogError(fmt.Sprintf("error when getting users map: got error %v", err))
+		log.Fatalf(fmt.Sprintf("error when getting users map: got error %v", err))
 	}
-	// Get authors of github pull request.
 	authors, err := prow.GetPrAuthorForPresubmit()
 	if err != nil {
 		if notPresubmit := prow.IsNotPresubmitError(err); *notPresubmit {
-			contextLogger.LogInfo(err.Error())
+			log.Infof(err.Error())
 		} else {
-			contextLogger.LogError(fmt.Sprintf("error when getting pr author for presubmit: got error %v", err))
+			log.Fatalf(fmt.Sprintf("error when getting pr author for presubmit: got error %v", err))
 		}
 	}
+
 	org, err := prow.GetOrgForPresubmit()
 	if err != nil {
 		if notPresubmit := prow.IsNotPresubmitError(err); *notPresubmit {
-			contextLogger.LogInfo(err.Error())
+			log.Infof(err.Error())
 		} else {
-			contextLogger.LogError(fmt.Sprintf("error when getting org for presubmit: got error %v", err))
+			log.Fatalf(fmt.Sprintf("error when getting org for presubmit: got error %v", err))
 		}
 	}
-	// TODO: move searching of user in to kymabot package
-	wg.Add(len(authors))
-	contextLogger.LogInfo(fmt.Sprintf("found %d authors in job spec env variable", len(authors)))
-	// Search entries for authors github usernames.
+
+	log.Infof(fmt.Sprintf("found %d authors in job spec env variable", len(authors)))
 	for _, author := range authors {
 		member, _, err := githubComClient.Organizations.IsMember(ctx, org, author)
 		if err != nil {
-			contextLogger.LogInfo(fmt.Sprintf("failed check if user %s is an github organisation member", author))
+			log.Fatalf(fmt.Sprintf("failed check if user %s is an github organisation member", author))
 		}
 		if member {
-			// Use goroutines.
-			go func(wg *sync.WaitGroup, author string, exitCode *atomic.Value) {
-				// Notify goroutine is done when exiting from it.
-				defer wg.Done()
-				for _, user := range usersMap {
-					if user.ComGithubUsername == author {
-						contextLogger.LogInfo(fmt.Sprintf("user %s is present in users map", author))
-						return
-					}
+			for _, user := range usersMap {
+				if user.ComGithubUsername == author {
+					log.Infof(fmt.Sprintf("user %s is present in users map", author))
+					break
 				}
-				contextLogger.LogError(fmt.Sprintf("user %s is not present in users map, please add user to users-map.yaml file.", author))
-				// Set exitcode to 1, to report failed prowjob execution.
-				exitCode.Store(1)
-			}(&wg, author, &exitCode)
-		} else {
-			wg.Done()
-		}
-	}
-	wg.Wait()
-	// If exitcode is nil, that means no errors were reported.
-	if exitCode.Load() == nil {
-		contextLogger.LogInfo("all authors present in users map or are not members of pull request github organisation")
-		err := contextLogger.Flush()
-		if err != nil {
-			fmt.Println(err.Error())
+			}
+			log.Fatalf(fmt.Sprintf("user %s is not present in users map, please add user to users-map.yaml file.", author))
```

Comment:
If you will have more than one PR author and `for` loop will not find match in `usersMap` for the first one, validation for other users will never happen. If next author will not be present in `usersMap`, an author will learn that on next prowjob run. With goroutines validation for all users will be executed on every prowjob execution.

Reply:
Reworked